addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -79,6 +81,7 @@ var SnippetEditor = Widget.extend({
         this.isTargetParentEditable = this.$target.parent().is(':o_editable');
         this.isTargetMovable = this.isTargetParentEditable && this.isTargetMovable;
         this.isTargetRemovable = this.isTargetParentEditable && !this.$target.parent().is('[data-oe-type="image"]');
+        this.isTargetReplaceable = this.$target.is('img');
```

Comment:
Don't make this part of the `SnippetEditor` class. The move and remove buttons have their own rules and are kinda the exception in the system. The replace button, is just a really specific option for images only: you can describe that with a simple option (data-js=...). Then, you can add this button in the overlay (you can be inspired by the `SnippetMove` option which does the same thing).

This would also make the feature extendable by the community and I am pretty we'll want that replace button for other things than images in the future ;)